content/post/88-js-ipfs-0.36.md

```diff
@@ -0,0 +1,102 @@
+---
+date: 2019-05-21
+url: 88-js-ipfs-0-36
+title: js-ipfs 0.36.0 released
+author: Alan Shaw
+---
+
+> URL safe CIDs, refs commands, DOM File support and more!
+
+# 🔦 Highlights
+
+## 🧬 Base32 encoding for v1 CIDs
+
+As a stepping stone towards switching to CIDv1 by default for all CIDs, we're shipping a small change to v1 CIDs that means their string form is `base32` encoded instead of `base58btc`. So, instead of:
+
+```console
+$ jsipfs add --cid-version 1 guardian.jpg
+added zb2rhk6GMPQF3hfzwXTaNYFLKomMeC6UXdUt6jZKPpeVirLtV guardian.jpg
+```
+
+You'll now get back a `base32` encoded CID:
+
+```console
+$ jsipfs add --cid-version 1 guardian.jpg
+added bafkreibu6pkzh33dfwfa3bg3twih7uiohu6d6cr34txljekdn3cvwoujiu guardian.jpg
+```
+
+You can [read more about this change here](https://github.com/ipfs/js-ipfs/issues/1995).
+
+PR: https://github.com/ipfs/js-ipfs/pull/2050
+
+## 👉 Added refs and refs local commands
+
+We've added the `refs` and `refs local` commands to the core, CLI and HTTP API. These commands allow you to list out all the CIDs referenced by a given DAG node or all the CIDs in your local repo. You can even choose the display format.
+
+PR: https://github.com/ipfs/js-ipfs/pull/2004
+
+## 🗄 Support for adding DOM File objects
+
+Finally! You can just add a [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) to IPFS without having to jump through hoops converting it to a `Buffer` first. Sensational!
+
+A `File`, by the way, is what you'll get back from a `<input type="file">` or the drag and drop API, so being able to easily take that file and add it to IPFS makes onboarding new frontend devs way easier (as well as way more convenient for all you old frontend devs 😜).
+
+PR: https://github.com/ipfs/js-ipfs/pull/2013
+
+## 🔬 MDNS discovery compatibility
+
+The libp2p discovery module for finding IPFS nodes over MDNS has been updated so that your JS IPFS node is now able to find Go IPFS nodes on the local network too. It's a backwards compatible change so older nodes will still be able to find your node and you'll still be able to find them.
+
+PR: https://github.com/libp2p/js-libp2p-mdns/pull/80
+
+## 🚤 28% faster stream multiplexing
+
+We switched the multiplexing implementation to one that's simpler, smaller and faster. We're estimating it to be [around 28% faster than the old implementation](https://github.com/libp2p/pull-mplex/pull/8#issue-254730751).
+
+PR: https://github.com/ipfs/js-ipfs/pull/1884
+
+## ⛩ Gateway improvements
+
+The IPFS HTTP gateway that JS IPFS exposes when run as a daemon in Node.js has been upgraded to support a number of HTTP features like conditional requests, byte range requests as well as getting some bug fixes, mini features and other improvements. There's too much to list here but check out the PR for and in depth description of what's changed.
+
+PR: https://github.com/ipfs/js-ipfs/pull/1989
```

Comment:
```suggestion
PR: https://github.com/ipfs/js-ipfs/pull/1989

## 🔏 Pubsub message signing

Messages sent over Pubsub will now be automatically signed by the initial publisher, and included as a signature property on each Pubsub message. IPFS and Libp2p will be verifying the signatures of Pubsub messages by default in future releases, which will enable us to verify the authenticity of all messages sent over the network before they are processed. Verification is not yet required by default, but it will be in future releases, and all unsigned messages will not be processed or forwarded.
```